### PR TITLE
Fix prediction with `num_beams > 1` and `num_outputs > 1`

### PR DESCRIPTION
**Pull request by s-jse**

### CODE REVIEW

File: genienlp/validate.py [hunk 1]
```diff
@@ -410,7 +410,7 @@ def validate(task, val_iter, model, numericalizer, args, num_print=10):
             output.predictions, output.answers, metrics_to_return, args.reduce_metrics, model.tgt_lang
         )
 
-        results = {'beam search': output.predictions, 'answer': output.answers, 'context': output.contexts}
+        results = {'model prediction': output.predictions, 'answer': output.answers, 'context': output.contexts}
```
Comment (Mehrad0711) [inline genienlp/validate.py]: Can we use "prediction" instead to be consistent with other keys?

File: genienlp/predict.py [hunk 2]
```diff
@@ -461,17 +466,26 @@ def run(args, device):
...
+                            + generation_output.answers[i]
+                        )]  # one line with all generation outputs separated by '\t'
                     if args.calibrator_paths is not None:
                         for score in generation_output.confidence_scores:
-                            line += '\t' + str(score[i])
-                    prediction_file.write(line + '\n')
+                            lines = [line + '\t' + str(score[i]) for line in lines] # append score to all lines
+                    prediction_file.write('\n'.join(lines) + '\n')
 
             if args.translate_return_raw_outputs:
```
Comment (Mehrad0711) [inline genienlp/predict.py:492]: Can we have "one_output_per_line" option for this if branch too?

File: genienlp/validate.py (hunk 1 above)
Comment (Mehrad0711) [inline genienlp/validate.py]: Also to make printed results align nicely in terminal, we use the maximum number of characters among all keys: https://github.com/stanford-oval/genienlp/blob/1d3f64bfa4d5077c5fe2748977d88ccdc16dd548/genienlp/validate.py#L392 
Maybe it's time to change the "11" to `max(len(key) for key in results)`

Comment (s-jse) [reply genienlp/validate.py]: I think "prediction" and "answer" are vague to first time users, as answer in some other code bases means the answer the model gives. So I changed them to "model prediction" and "gold answer".

Comment (Mehrad0711) [reply genienlp/validate.py]: yeah, this is another one of those things we need to standardize throughout the code. We have "beam search" here and "greedy" in metrics.py!
looks good for this PR, but should we go with "pred" and "gold" everywhere?

File: genienlp/predict.py [hunk 3]
```diff
@@ -490,14 +490,23 @@ def run(args, device):
...
                     for i in range(len(generation_output.example_ids)):
-                        line = (
-                            generation_output.example_ids[i]
-                            + '\t'
-                            + '\t'.join(generation_output.raw_predictions[i])
-                            + '\t'
-                            + generation_output.answers[i]
-                        )  # all outputs separated by '\t'
-                        prediction_file.write(line + '\n')
+                        if args.one_output_per_line:
```
Comment (Mehrad0711) [inline genienlp/predict.py:495]: Thanks. I'll remove this branching thing too. We should add a function for creating the correct output.

**Review by Mehrad0711 — APPROVED** [review]

**Pull request merged**